### ISSUE TITLE
Typescript Typings are out of date

### ISSUE THREAD

**Issue by pradyuman** [kriasoft/universal-router#101]
Is it possible for you guys to update the existing typings for Typescript?
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e5ed4fc91679670a415f2534005ef8d67ec2189e/types/universal-router/index.d.ts

There were breaking API changes in v3 so it would be nice to have the typings reflect those.

Thanks!

**Comment by dangmai** [comment]
I'm trying to update the typing for `universal-router` to the newest version, and there's an annoying issue I'd like to bring up. Right now the distributed library is a CommonJS module, not an ES6 module, and thus we can't do something like this in Typescript:

```
import Router from 'universal-router';
```

We have to do this instead:

```
import Router = require('universal-router');
```

This is because a CommonJS module exports the Router class instead of a `default` that points to the Router class (which is what Typescipt expects). Note that a transpiler like Babel does some interop magic behind the scene to accommodate this (see https://github.com/Microsoft/TypeScript/issues/11179) which Typescript currently does not (proposal to fix this is tracked at https://github.com/Microsoft/TypeScript/issues/16093)

Is there a way for `universal-router` to export a module that is compliant with both CommonJS and ES6 modules? Here's how some other libraries are doing it - https://github.com/Microsoft/TypeScript/issues/16093#issuecomment-304589484

**Comment by frenzzy** [comment]
Maybe `"module": "main.mjs"` section in `package.json` could solve this?

**Comment by anantoghosh** [comment]
@dangmai You can use the allowSyntheticDefaultImports compiler option in your tsconfig
```js
{
  "compilerOptions": {
    "allowSyntheticDefaultImports": true
    ...

```

**Comment by frenzzy** [comment]
Release of v4 is coming soon, would be nice to release it with typescript definitions. Could someone help with it for version in master branch? maybe @jtmthf, thanks in advance

**Comment by frenzzy** [comment]
[Flow](https://flow.org/) types: https://github.com/flowtype/flow-typed/pull/1596

**Comment by gytisgreitai** [comment]
Any work around ? the `import Router = require('universal-router');` does not seem to work with ts 2.6.x

`Import assignment cannot be used when targeting ECMAScript modules. Consider using 'import * as ns from "mod"', 'import {a} from "mod"', 'import d from "mod"', or another module format instead.`

**Comment by frenzzy** [comment]
@gytisgreitai can you try `require('universal-router/main')` ?

**Comment by gytisgreitai** [comment]
no, same error. `let UniversalRouter = require('universal-router/main');` just sets UniversalRouter to a js file.

I'm using react-creat-app-ts so maybe there is something in between. For now I've simply resorted to unpkg since this is a small intranet project.
